src/EditorFeatures/Core.Wpf/InlineRename/InlineRenameSession.cs

```diff
@@ -297,7 +299,8 @@ private void UpdateReferenceLocationsTask(Task<IInlineRenameLocationSet> allRena
                     await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, _cancellationTokenSource.Token);
                     _cancellationTokenSource.Token.ThrowIfCancellationRequested();
 
-                    RaiseSessionSpansUpdated(t.Result.Locations.ToImmutableArray());
+                    var locations = _workspace.IgnoreUnchangeableDocumentsWhenApplyingChanges ? t.Result.Locations.Where(l => l.Document.CanApplyChange()) : t.Result.Locations;
```

Comment:
> Good job catching that inline rename still needs to filter instead of potentially applying temporary edits to buffers. But if there's a conflict resolution in a file that otherwise didn't have a original location, I suspect we're going to try modifying the text buffer too. We may need to move the check you added to somewhere else. Or if this does work, a test might be good.

Good catch. Turns out this change only affect the text shown in the UI, not the actual buffer changes. i.e. "rename will update x references in y files", and if you have conflicts in doc w/o reference, the conflict text would still be incorrect (although this fix would still work in the sense that it won't crash and changes will be dropped silently by workspace). At this point, I'm inclining to undo this change here and make the complete fix for inline rename in a separate PR (possibly 16.6 depends on potential impact) for risk reduction. Thoughts?